odoo/addons/base/populate/res_company.py

```diff
@@ -19,10 +19,11 @@ class Partner(models.Model):
     def _populate_factories(self):
         # Activate currency to avoid fail iterator
         (self.env.ref('base.USD') | self.env.ref('base.EUR')).active = True
+        last_id = self.env["res.company"].search([], order="id desc", limit=1).id
 
         # remaining: paperformat_id, parent_id, partner_id, font, report_header, external_report_layout_id, report_footer
         def get_name(values=None, counter=0, **kwargs):
-            return 'company_%s_%s' % (counter, self.env['res.currency'].browse(values['currency_id']).name)
```

Comment:
Note: just making populate of base models re-entrant to significantly improve its usage and debugging